internal/controller/scheduler/completions.go

```diff
@@ -61,14 +61,14 @@ func (w *completionsWatcher) OnUpdate(old interface{}, new interface{}) {
 func (w *completionsWatcher) cleanupSidecars(pod *v1.Pod) {
 	if terminated := getTermination(pod); terminated != nil {
 		if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-			job, err := w.k8s.BatchV1().Jobs(pod.Namespace).Get(context.TODO(), pod.Labels["job-name"], metav1.GetOptions{})
+			ctx := context.Background()
```

Comment:
Another thing for a "cleanup TODOs" list... until then I would leave it as `context.TODO()`

```suggestion
			ctx := context.TODO()
```